include/dsn/internal/task.h

```diff
@@ -118,6 +120,7 @@ class task :
     bool                    cancel(bool wait_until_finished, /*out*/ bool* finished = nullptr);
     bool                    wait(int timeout_milliseconds = TIME_MS_MAX, bool on_cancel = false);
     virtual void            enqueue();
+    bool                    set_retry(bool enqueue_immediately = true); // return true when called inside exec(), false otherwise
```

Comment:
I think the comment is not exact, maybe "return true if set succeed inside exec(), false otherwise" is better